python/cudf/cudf/core/dataframe.py

```diff
@@ -6343,6 +6343,18 @@ def wrapper(self, other, axis="columns", level=None, fill_value=None):
             raise NotImplementedError("Only axis=1 supported at this time.")
         return wrapped_func(self, other, axis, level, fill_value)
 
+    # The above has wrapped binary op function from `Frame` with `wrapper`,
+    # copied module level attributes to wrapper (`__docs__` etc.) and set
+    # `__wrapped__` attribute of `wrapper` to `Frame` binop. Since this wrapper
+    # reordered the function arguments, we need to make sure the function
+    # signature in sphinx also matches. `inspect.signature` tells that
+    # if a function has `__wrapped__` attribute defined, the signature of that
+    # will overwrite the signature from the original function. Below utilizes
+    # this rule to make sure sphinx is rendering the right argument list for
+    # the function.
+    wrapper.__wrapped__ = (
+        lambda self, other, axis="columns", level=None, fill_value=None: None
```

Comment:
If replacing the signature is what is desired, should we update `wrapper.__signature__` instead? Disclaimer: I'm fuzzy on the details of both `__wrapper__` and `__signature__`, but just mentioning it here if it's a more appropriate solution.

Reply:
`functools.wraps` creates the `__wrapper__` attribute regardless. When we add the `__signature__` attribute, conceptually there's a dilemma where to get the signature source from. In [cpython](https://github.com/python/cpython/blob/3f8b23f8ddab75d9b77a3997d54e663187e12cc8/Lib/inspect.py#L2376-L2377) it seems like it will traverse down `__wrapped__` until it hits the first wrapped function with `__signature__` defined (or otherwise return the last wrapped func). 

In this case the above suggested works because the top level function is set with the desired `__signature__`. But may be a bit brittle because I'm not sure if the above logic is documented anywhere. An alternative is to set `__signature__` attribute of `__wrapped__` with `inspect.replace`. Though this may seem a bit tedious.

Reply:
The `__wrapped__` attribute was introduced in Python 3.5, specifically in [PEP 362](https://www.python.org/dev/peps/pep-0362/) for the sole purpose of signature introspection using the `inspect` module. To my knowledge this is the only usage of the attribute. That said, I'm not opposed to instead overwriting `__signature__` since that's certainly a more targeted solution for what we're trying to achieve.